### ISSUE TITLE
virtual.testcloud asks for password if `--user` is used

### ISSUE THREAD

**Issue by lukaszachy** [teemtee/tmt#1452]
Reproducer: `tmt run provision -h virtual -u demo login finish`

In log.txt it is visible that somewhere str got iterated over its characters (should be -i/var/tmp/tmt/run-365/plans/default/provision/default/id_ecdsa) :
```
06:12:15         Execute command 'whoami' on guest '127.0.0.1'.
06:12:15         Create the master ssh connection: ssh -oForwardX11=no -oStrictHostKeyChecking=no -oUserKnownHostsFile=/dev/null -oIdentitiesOnly=yes -p10041 -i/ -iv -ia -ir -i/ -it -im -ip -i/ -it -im -it -i/ -ir -iu -in -i- -i3 -i6 -i5 -i/ -ip -il -ia -in -is -i/ -id -ie -if -ia -iu -il -it -i/ -ip -ir -io -iv -ii -is -ii -io -in -i/ -id -ie -if -ia -iu -il -it -i- -i0 -i/ -ii -id -i_ -ie -ic -id -is -ia -S/run/user/17062/tmt/tmp_sqvs1xr -MNnT demo@127.0.0.1
```
However even with correct `-i ` value the identity file does not work.

Cloud-init at https://github.com/teemtee/tmt/blame/main/tmt/steps/provision/testcloud.py#L44 doesn't assign public_key to the correct user (password works and ~demo/.authorized_keys file is empty)

tmt doesn't modify the password so [testcloud's](https://pagure.io/testcloud/blob/master/f/testcloud/config.py#_107) one "passw0rd" is set.

**Comment by happz** [comment]
I can confirm the password question, but, since I was worried about "factorization" of the key path, I tried to reproduce, but key path in my log looks correct:

```
06:32:49         Execute command 'whoami' on guest '127.0.0.1'.
06:32:49         Create the master ssh connection: ssh -oForwardX11=no -oStrictHostKeyChecking=no -oUserKnownHostsFile=/dev/null -oIdentitiesOnly=yes -p10023 -i/var/tmp/tmt/run-074/plans/test/provision/default/id_ecdsa -S/run/user/1000/tmt/tmp0xfqezou -MNnT demo@127.0.0.1
```

Your logs suggest a single key has been given, somewhere, somehow, where `key` parameter is expected to be a list of string. And this is actually exactly what happens, see https://github.com/teemtee/tmt/blob/main/tmt/steps/provision/testcloud.py#L462, `self.key` comes from `GuestSsh`, and it's `key: List[str]` - which raises the question, how the hell it's working for me... :/

**Comment by happz** [comment]
Right, it worked, because I've been using `tmt` from an rpm, not having the virtualenv active. Now it makes sense, and there is a bug in testcloud plugin's use of keys.

**Comment by lukaszachy** [comment]
Wow. Even without user it mallforms the key - so two issues for the price of one.

**Comment by happz** [comment]
Yeah. Python is nice, but once the project grows, moving bits around without stricter control like annotations results in exactly this :/

Anyway, here's possible fix, according to logs, key was used correctly: https://github.com/teemtee/tmt/pull/1453

**Comment by psss** [comment]
Seems to be fixed by #1453.
